pkg/core/client.go

```diff
@@ -62,8 +70,17 @@ type client struct {
 	eventing     eventing_cs.Interface
 	serving      serving_cs.Interface
 	clientConfig clientcmd.ClientConfig
+	kubeCtl      kubectl.KubeCtl
+	kustomizer   kustomize.Kustomizer
 }
 
-func NewClient(clientConfig clientcmd.ClientConfig, kubeClient kubernetes.Interface, eventing eventing_cs.Interface, serving serving_cs.Interface) Client {
-	return &client{clientConfig: clientConfig, kubeClient: kubeClient, eventing: eventing, serving: serving}
+func NewClient(clientConfig clientcmd.ClientConfig, kubeClient kubernetes.Interface, eventing eventing_cs.Interface, serving serving_cs.Interface, kubeCtl kubectl.KubeCtl, kustomizer kustomize.Kustomizer) Client {
+	return &client{
```

Comment:
maybe this PR could be an opportunity to rename `client` to a more descriptive, less vague name?

Reply:
Ultimately, I think we should decompose 'client' into a set of domain specific clients. I don't want to do that right now though.

Reply:
👍